compiler/tests/fixtures/demos/uniswap.fe

```diff
@@ -0,0 +1,365 @@
+        amount0: u256
+        amount1: u256
+        idx to: address
+
+    event Swap:
+        idx sender: address
+        amount0_in: u256
+        amount1_in: u256
+        amount0_out: u256
+        amount1_out: u256
+        idx to: address
+
+    event Sync:
+        reserve0: u256
+        reserve1: u256
+
+    pub def __init__():
+        self.factory = msg.sender
+
+    pub def factory() -> address:
+        return self.factory
+
+    pub def token0() -> address:
+        return self.token0
+
+    pub def token1() -> address:
+        return self.token1
+
+    def _mint(to: address, value: u256):
+        self.total_supply = self.total_supply + value
+        self.balances[to] = self.balances[to] + value
+        emit Transfer(address(0), to, value)
+
+    def _burn(from: address, value: u256):
+        self.balances[from] = self.balances[from] - value
+        self.total_supply = self.total_supply - value
+        emit Transfer(from, address(0), value)
+
+    def _approve(owner: address, spender: address, value: u256):
+        self.allowances[owner][spender] = value
+        emit Approval(owner, spender, value)
+
+    def _transfer(from: address, to: address, value: u256):
+        self.balances[from] = self.balances[from] - value
+        self.balances[to] = self.balances[to] + value
+        emit Transfer(from, to, value)
+
+    pub def approve(spender: address, value: u256) -> bool:
+        self._approve(msg.sender, spender, value)
+        return true
+
+    pub def transfer(to: address, value: u256) -> bool:
+        self._transfer(msg.sender, to, value)
+        return true
+
+    pub def transferFrom(from: address, to: address, value: u256) -> bool:
+        assert self.allowances[from][msg.sender] >= value
+
+        self.allowances[from][msg.sender] = self.allowances[from][msg.sender] - value
+        self._transfer(from, to, value)
+        return true
+
+    pub def balanceOf(account: address) -> u256:
+        return self.balances[account]
+
+    # TODO: support structs in the public interface (https://github.com/ethereum/fe/issues/282)
+    pub def get_reserves(): # -> ThreeNums:
+#        return ThreeNums(
+#            num1=self.reserve0,
+#            num2=self.reserve1,
+#            num3=self.block_time_stamp
+#        )
+        pass
+
+    # called once by the factory at time of deployment
+    pub def initialize(token0: address, token1: address):
+        assert msg.sender == self.factory, "UniswapV2: FORBIDDEN"
+        self.token0 = token0
+        self.token1 = token1
+
+    # update reserves and, on the first call per block, price accumulators
+    def _update(balance0: u256, balance1: u256, reserve0: u256, reserve1: u256):
+        #assert balance0 <= u112(-1) and balance1 <= u112(-1), "UniswapV2: OVERFLOW"
+        # changed from u32s
+        block_timestamp: u256 = block.timestamp % 2**32
+        time_elapsed: u256 = block_timestamp - self.block_timestamp_last # overflow is desired
```

Comment:
>`# overflow is desired`

If the overflow is desired than I think this would eventually have to use something like `wrapping(block_timestamp) - wrapping(self.block_timestamp_last)` since Fe will not allow checked arithmetic by default. This is mentioned here https://github.com/ethereum/fe/issues/153 and is inspired by Rust's [Wrapping](https://doc.rust-lang.org/std/num/struct.Wrapping.html) type. I think it shouldn't be too hard to implement.

Reply:
it seems like there is some nuanced behavior here. Im not quite sure why these overflows are desired..

for the time being, we could also consider just using `%` to reproduce the "desired" overflow

Reply:
But we will soon have checks for all arithmetic operations. So, we need to think about how that would work in the future. If you like the [Wrapping](https://doc.rust-lang.org/std/num/struct.Wrapping.html)  approach, I guess we could implement that fairly quickly. I would imagine we simply add another type `WrappedInteger` similar to `Integer` and a type constructor `wrapped` that e.g. produced as `WrappedInteger::U16` when you call `wrapped(some_u16_val)`. And the mapper would simply map these to the unchecked arithmetic opcodes. How does that sound to you?

Reply:
Im not opposed to adding a wrapped type https://github.com/ethereum/fe/issues/286